src/replication.c

```diff
@@ -1189,6 +1189,24 @@ send_replica_snapshot(spec_t *spec, replica_t *replica, uint64_t io_seq,
 	return ret;
 }
 
+static void
+disconnect_pending_replica(replica_t *replica, uint64_t io_seq,
+    zvol_op_code_t opcode)
+{
+	mgmt_cmd_t *mgmt_cmd;
+
+	MTX_LOCK(&replica->r_mtx);
+	TAILQ_FOREACH(mgmt_cmd, &replica->mgmt_cmd_queue, mgmt_cmd_next) {
+		if (mgmt_cmd->io_hdr->io_seq == io_seq &&
+		    mgmt_cmd->io_hdr->opcode == opcode) {
+			inform_mgmt_conn(replica);
+			break;
```

Comment:
we may need to continue the loop

Reply:
not needed, as disconnect_conn is replica level flag. So we can break from here.